IPython/terminal/embed.py

```diff
@@ -83,6 +96,8 @@ def __init__(self, **kw):
 
         super(InteractiveShellEmbed,self).__init__(**kw)
 
+        frame = sys._getframe(InteractiveShellEmbed._stack_frames)
+        self._call_location_id = '%s:%s' % (frame.f_code.co_filename, frame.f_lineno)
```

Comment:
Yes that's why the PR description says: 

> The frame inspection is by default to 3, which is the correct number of
frame when using the Import IPython; IPython.embed() syntax, it will
be the wrong number of frames when creating an InteractiveShellEmbed
manually, leading to kill_embed to likely not work correctly.
Though I expect these case to be rare, and people doing so to be capable
of wrapping their custom call in a context_manager that set
InteractiveShellEmbed._stack_frames to the right value depending on
the context.